File: easy/3120_count_the_number_of_special_characters_i/solution.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        case_map = [0] * 26

        special_chars_count = 0

        for c in word:
            if c.isupper():
                c_i = ord(c) - 65
                case_map[c_i] |= 1
            else:
                c_i = ord(c) - 97
                case_map[c_i] |= 2

            if case_map[c_i] == 3:
                case_map[c_i] = 7
                special_chars_count += 1

        return special_chars_count

File: easy/3120_count_the_number_of_special_characters_i/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  int numberOfSpecialChars(std::string word) {
    std::vector<int> caseMap(26, 0);

    int specialCharsCount = 0;

    for (const char &ch : word) {
      size_t chI;

      if (isupper(ch)) {
        chI = ch - 65;
        caseMap[chI] |= 1;
      } else {
        chI = ch - 97;
        caseMap[chI] |= 2;
      }

      if (caseMap[chI] == 3) {
        caseMap[chI] = 7;
        ++specialCharsCount;
      }
    }

    return specialCharsCount;
  }
};
